File: Sources/Shared/Websocket/Packet/Bound/Client/PacketClientboundConnectionKickReason.ts
export const enum PacketClientboundConnectionKickReason {
    OutdatedClient,
    CheatDetected,
}

type DisplayConnectionKickReason = Capitalize<string>;

export const DISPLAY_KICK_REASON = {
    [PacketClientboundConnectionKickReason.OutdatedClient]: "Outdated client",
    [PacketClientboundConnectionKickReason.CheatDetected]: "Cheat detected",
} satisfies Record<PacketClientboundConnectionKickReason, DisplayConnectionKickReason>;

File: Sources/server/internal/network/packet_clientbound_connection_kick_reason.go
package network

type ConnectionKickReason uint

const (
	OutdatedClient ConnectionKickReason = iota
	CheatDetected
)

var DisplayKickReasons = map[ConnectionKickReason]string{
	OutdatedClient: "Outdated client",
	CheatDetected: "Cheat detected",
}